include/aspect/particle/property/elastic_tensor_decomposition.h

```diff
@@ -0,0 +1,210 @@
+           * beginning of the program after parse_parameters is run.
+           */
+          void
+          initialize () override;
+
+          /**
+           * Initialization function. This function is called once at the
+           * creation of every particle for every property to initialize its
+           * value.
+           *
+           * @param [in] position The current particle position.
+           * @param [in,out] particle_properties The properties of the particle
+           * that is initialized within the call of this function. The purpose
+           * of this function should be to extend this vector by a number of
+           * properties.
+           */
+          void
+          initialize_one_particle_property (const Point<dim> &position,
+                                            std::vector<double> &particle_properties) const override;
+
+          /**
+           * Update function. This function is called every time an update is
+           * request by need_update() for every particle for every property.
+           *
+           * @param [in] data_position. An unsigned integer that denotes which
+           * component of the particle property vector is associated with the
+           * current property. For properties that own several components it
+           * denotes the first component of this property, all other components
+           * fill consecutive entries in the @p particle_properties vector.
+           *
+           * @param [in] position. The current particle position.
+           *
+           * @param [in] solution. The values of the solution variables at the
+           * current particle position.
+           *
+           * @param [in] gradients. The gradients of the solution variables at
+           * the current particle position.
+           *
+           * @param [in,out] particle_properties. The properties of the particle
+           * that is updated within the call of this function.
+           */
+          void
+          update_one_particle_property (const unsigned int data_position,
+                                        const Point<dim> &position,
+                                        const Vector<double> &solution,
+                                        const std::vector<Tensor<1,dim>> &gradients,
+                                        const ArrayView<double> &particle_properties) const  override;
+
+          /**
+           * This function tells the particle manager that
+           * we need to update particle properties.
+           */
+          UpdateTimeFlags
+          need_update () const override;
+
+          /**
+           * Return an even permutation based on an index. This is an internal
+           * utility function, also used by the unit tester.
+           */
+          static
+          std::array<unsigned short int, 3>
+          indexed_even_permutation(const unsigned short int index);
```

Comment:
As discussed can you move the static members and functions into a `Utility` namespace outside of this class?